docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
+
+The features going to be implement:
+  testcases for generating long problem sets questions and check answer, as well as the testcases for UI, Checker, and Storage.
+
+Details for implementation:
+1. Testcases for generating long problem sets questions:
+   implement specific input and call assertTrue to verify if the generated long problem sets has the desired format.
+2. Testcases for checker :
+   provide answer to sample testcases and verify if the checker output the accurate correctness.
+3. Testcases for UI:
+   verify the UI output's correctness.
+4. Storage:
+   verify if the record file is created, and the content in the file is correct.
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+* The student who wish to practise their mathematical problem solving ability.  
+* The teachers who wish to provide students with some math problems and check their performance.  
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+The product automates the generation of mathematical problems and their corresponding answers, thereby reducing the effort required by students to find practice problems.
 
 ## User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
 |--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|student who just started learning quadratic equations|use MathGenius to specifically generate problem sets|gain specialized practice on this topic|
+|v1.0|a student who’s unfamiliar with mathematics terms|watch explanations and introductions to unfamiliar terms|strengthen understanding|
+|v1.0|primary school teacher|wcreate course or topic-specific arithmetic questions for students|teaching student more effortlessly|
+|v2.0|student who want to pracise solving foumula|generate various kinds of formula|practise the ability of solving math formula|
 
 ## Non-Functional Requirements
-
-{Give non-functional requirements}
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.  
+2. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.   
+    
+**{More to be add}**
 
 ## Glossary
 
-* *glossary item* - Definition
+* Mainstream OS: Windows, Linux, Unix, MacOS  
+* Private contact detail: A contact detail that is not meant to be shared with others  
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+## Launch and Shutdown
+   1. initial launch:
+      1. Download the jar file and copy into an empty folder
+      2. run the jar file use command '_java -jar tp.main.jar_'
```

Comment:
It might be more readable to use markdown code block for '_java -jar tp.main.jar_'